infra/cifuzz/cifuzz.py

```diff
@@ -190,14 +190,15 @@ def run_fuzzers(fuzz_seconds, workspace, project_name):
     logging.error('No fuzzers were found in out directory: %s.',
                   format(out_dir))
     return False, False
-  fuzz_seconds_per_target = fuzz_seconds // len(fuzzer_paths)
 
   # Run fuzzers for alotted time.
+  fuzzers_left = len(fuzzer_paths)
   for fuzzer_path in fuzzer_paths:
-    target = fuzz_target.FuzzTarget(fuzzer_path, fuzz_seconds_per_target,
-                                    out_dir, project_name)
+    run_seconds = fuzz_seconds // fuzzers_left
+    target = fuzz_target.FuzzTarget(fuzzer_path, run_seconds, out_dir,
+                                    project_name)
 
-    test_case, stack_trace = target.fuzz()
+    test_case, stack_trace, elapsed_time = target.fuzz()
```

Comment:
why not make the `fuzz` interface simpler, and do the timing logic in this function instead (which seems simpler overall)? it seems like the only usecase would be here.

i.e. something like:

```python


expected_end_time = time.time() + fuzz_seconds
for i, fuzzer_path in enumerate(fuzzer_paths):
  seconds_remaining = expected_end_time - time.time()
  if seconds_remaining <= 0:
    break
  fuzzers_left = len(fuzzer_paths) - i
  run_seconds = seconds_remaining // fuzzers_left
  ...
```

Reply:
Hmm, I'm confused -- probably missing something here. The place I'm proposing to have this is in this function (cifuzz.py / run_fuzzers), not in the fuzz target? This seems like a better place for timing how long has passed, and is used directly here. This seems better than delegating that to fuzz(). Is there some future changed planned here?

Reply:
I'm also not sure why they would run for different amounts of time. Maybe it's if one finds a bug we don't want to report (e.g. repros on old builds, timeouts or ooms maybe) so we can spend the time on other fuzzers.

Reply:
@oliverchang Sorry I misunderstood your comment. That makes a lot more sense now. I will implement the change.

Reply:
Fixed.